### PR TITLE
Fix for non-finite gamma

### PR DESCRIPTION
**Pull request by korken89**
In my search for the following issue (https://github.com/korken89/lbfgs-rs/pull/8) I also found what is in this PR.

Is there a better way to estimate gamma when there is no curvature?
Right now I just made it based on the starting cost and gradient, which works well in my tests but I don't think its a good solution.

### CODE REVIEW

File: src/core/panoc/panoc_engine.rs [hunk 1]
```diff
@@ -350,7 +350,15 @@ where
         self.cache.reset();
         (self.problem.cost)(u_current, &mut self.cache.cost_value)?; // cost value
         self.estimate_loc_lip(u_current)?; // computes the gradient as well! (self.cache.gradient_u)
-        self.cache.gamma = GAMMA_L_COEFF / self.cache.lipschitz_constant;
+        let gamma = GAMMA_L_COEFF / self.cache.lipschitz_constant;
+        self.cache.gamma = if gamma.is_finite() {
```
[the remark this answers is not in the document]
Comment (korken89) [reply src/core/panoc/panoc_engine.rs:354]: Hi, that's also an acceptable solution to me!
I added a test where this problem is shown.

[the remark this answers is not in the document]
Comment (korken89) [reply src/core/panoc/panoc_engine.rs:354]: I moved to your fix, I see similar performance in solution quality and speed.

**Comment by korken89** [comment]
This issue only seems to happen when you have one uses L1 norm or robust cost functions.
This popped up for me when I was using the Huber norm, and got a lot of outliers in the data.

I remember we discussed to some length before how PANOC could better support problems which have a L1 norm in it, did we ever write out thoughts down somewhere on that?

**Review by alphaville — APPROVED** [review]

**Comment by korken89** [comment]
Nice additions, thanks for cleaning up my mess! xD

**Pull request merged**